gcloud/_helpers.py

```diff
@@ -217,6 +235,9 @@ def _determine_default_project(project=None):
         project = _get_production_project()
 
     if project is None:
+        project = _file_project_id()
```

Comment:
By inserting it here, you match the order of precedence I am suggesting, so only the docs need to change to match the actual order the Project ID is checked.